### ISSUE TITLE
Calling a method should reset the nullable state for members of the receiver

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29975]
```
        // Calling a method should reset the state for members.
        [Fact]
        public void Members_CallMethod()
        {
            var source =
@"#pragma warning disable 0649
class A
{
    internal object? P { get; set; }
}
class B
{
    internal A? Q { get; set; }
}
class Program
{
    static void M()
    {
        object o;
        B b = new B() { Q = new A() { P = new object() } };
        o = b.Q.P; // 1
        b.Q.P.ToString();
        o = b.Q.P; // 2
        b.Q.ToString();
        o = b.Q.P; // 3
        b = new B() { Q = new A() { P = new object() } };
        o = b.Q.P; // 4
        b.ToString();
        o = b.Q.P; // 5
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            // PROTOTYPE(NullableReferenceTypes): Should report warnings.
            comp.VerifyDiagnostics(/*...*/);
        }
```

**Comment by jcouv** [comment]
I thought LDM had explicitly ruled this out, but digging through the LDM notes and Mads' spec I couldn't find a record of that decision.

**Comment by jcouv** [comment]
@cston confirmed my understanding and pointed out that Mads referenced this behavior in LDM yesterday (we assume that method calls don't change the world outside of our sight).
I'll go ahead and close.
